wireapp/wire-android
# feat(calling): end call on member leave or conversation deleted (WPB-2955)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-2955" title="WPB-2955" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-2955</a>  [android] Call not ended, when being removed from an conversation
  </td></table>
  <br />
 


























----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Call not ended, when being removed from a conversation
 
### Causes (Optional)

Not implemented 

### Solutions

I used `EndCallOnConversationChangeUseCase`  in `GlobalObserversManager` to keep observing active call and it's conversation details in background and foreground.

The moment the user is no longer part of that conversation, conversation is deleted or the user is deleted, the call will be terminated


### Dependencies (Optional)

Needs this PR
[feat(calling): create usecase for ending call on member leave or conversation delete (WPB-2955)
](https://github.com/wireapp/kalium/pull/1865)

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

- Be in a group call
- Remove you self from a different device or current device
--> Call should be ended

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 



----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
